### PR TITLE
fix(titanium-loading-indicator): ensure veil is closed when closing is done

### PR DESCRIPTION
**Pull request by AbandonedFridge**
Should fix issue where sometimes pages fully load but the veil doesn't go away.
https://devpcms.leavitt.com/sales-expectation/10934/expectations seems to trigger the issue often

### CODE REVIEW

**Review by caldjoy — APPROVED** [review]

**Review by SketchBookkeeper — APPROVED** [review]

**Review by reese-mitchell — APPROVED** [review]

**Review by AwsomeWyatt — APPROVED** [review]

**Review by ElijahWilliamsLSS — APPROVED** [review]

File: packages/titanium-loading-indicator/src/titanium-full-page-loading-indicator.ts [hunk 1]
```diff
@@ -79,6 +79,7 @@ export class TitaniumFullPageLoadingIndicatorElement extends LitElement {
       this._animationTimer = window.setTimeout(() => {
         this.handleAnimationTimerEnd_();
         document.body.style.overflow = '';
+        this.opened = false;
```
Comment (aarondrabeck) [inline packages/titanium-loading-indicator/src/titanium-full-page-loading-indicator.ts:82]: This is set to false already on line 76

File: packages/titanium-loading-indicator/src/titanium-full-page-loading-indicator.ts [hunk 2]
```diff
@@ -66,7 +66,7 @@ export class TitaniumFullPageLoadingIndicatorElement extends LitElement {
   private _close() {
     window.clearTimeout(this._openDelayTimer);
     const totalTimeOpened = performance.now() - this._timeOpen;
-    const closeDelay = totalTimeOpened > this._minTimeOpen ? 0 : this._minTimeOpen - totalTimeOpened || 0;
+    const closeDelay = Math.max(this._minTimeOpen - totalTimeOpened, 0);
```
Comment (aarondrabeck) [inline packages/titanium-loading-indicator/src/titanium-full-page-loading-indicator.ts:69]: Is there a functional difference here?

Comment (AbandonedFridge) [reply packages/titanium-loading-indicator/src/titanium-full-page-loading-indicator.ts:69]: Math.max returns the bigger of the two numbers. It's functionally identical, I just find the way it was before to be difficult to read.

File: packages/titanium-loading-indicator/src/titanium-full-page-loading-indicator.ts (hunk 1 above)
Comment (AbandonedFridge) [reply packages/titanium-loading-indicator/src/titanium-full-page-loading-indicator.ts:82]: Yes. But a weird race condition with the render cycle makes it so sometimes it doesn't. This is just to ensure that it gets cleaned up at the end. On https://devpcms.leavitt.com/sales-expectation/10934/expectations with the loading veil as is, I was getting stuck on loading veil about 1 in 15 times I reload the page. With this, I wasn't able to get stuck in the veil.

File: packages/titanium-loading-indicator/src/titanium-full-page-loading-indicator.ts (hunk 2 above)
Comment (AbandonedFridge) [reply packages/titanium-loading-indicator/src/titanium-full-page-loading-indicator.ts:69]: I'm saying, I don't actually care about this. I'm happy to revert it if you like the other way better.

Comment (aarondrabeck) [reply packages/titanium-loading-indicator/src/titanium-full-page-loading-indicator.ts:69]: sounds good just checking I wasn't missing something, I like the simplicity in this change

File: packages/titanium-loading-indicator/src/titanium-full-page-loading-indicator.ts (hunk 1 above)
Comment (aarondrabeck) [reply packages/titanium-loading-indicator/src/titanium-full-page-loading-indicator.ts:82]: I think I see the issue here.  The the runNextAnimation is sometimes running after close was initiated.  

if we add to line 58 ``` if(this.closing) return; ``` , it should prevent this from occurring.

**Review by aarondrabeck — APPROVED** [review]

**Pull request merged**